extensions-core/namespace-lookup/src/main/java/io/druid/server/lookup/namespace/NamespacedExtractionModule.java

```diff
@@ -0,0 +1,106 @@
+
+import com.fasterxml.jackson.databind.Module;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.google.common.collect.ImmutableList;
+import com.google.inject.Binder;
+import com.google.inject.Key;
+import com.google.inject.TypeLiteral;
+import com.google.inject.multibindings.MapBinder;
+import io.druid.guice.LazySingleton;
+import io.druid.guice.PolyBind;
+import io.druid.initialization.DruidModule;
+import io.druid.query.lookup.NamespaceLookupExtractorFactory;
+import io.druid.query.lookup.namespace.ExtractionNamespace;
+import io.druid.query.lookup.namespace.ExtractionNamespaceCacheFactory;
+import io.druid.query.lookup.namespace.JDBCExtractionNamespace;
+import io.druid.query.lookup.namespace.StaticMapExtractionNamespace;
+import io.druid.query.lookup.namespace.URIExtractionNamespace;
+import io.druid.server.lookup.namespace.cache.NamespaceExtractionCacheManager;
+import io.druid.server.lookup.namespace.cache.OffHeapNamespaceExtractionCacheManager;
+import io.druid.server.lookup.namespace.cache.OnHeapNamespaceExtractionCacheManager;
+import java.util.List;
+
+/**
+ *
+ */
+public class NamespacedExtractionModule implements DruidModule
+{
+  public static final String TYPE_PREFIX = "druid.lookup.namespace.cache.type";
+
+  @Override
+  public List<? extends Module> getJacksonModules()
+  {
+    return ImmutableList.<Module>of(
+        new SimpleModule("DruidNamespacedCachedExtractionModule")
+            .registerSubtypes(
+                NamespaceLookupExtractorFactory.class
+            )
+    );
+  }
+
+  public static MapBinder<Class<? extends ExtractionNamespace>, ExtractionNamespaceCacheFactory<?>> getNamespaceFactoryMapBinder(
+      final Binder binder
+  )
+  {
+    return MapBinder.newMapBinder(
+        binder,
+        new TypeLiteral<Class<? extends ExtractionNamespace>>()
+        {
+        },
+        new TypeLiteral<ExtractionNamespaceCacheFactory<?>>()
+        {
+        }
+    );
+  }
+
+  @Override
+  public void configure(Binder binder)
+  {
+    PolyBind.createChoiceWithDefault(
+        binder,
+        TYPE_PREFIX,
+        Key.get(NamespaceExtractionCacheManager.class),
+        Key.get(OnHeapNamespaceExtractionCacheManager.class),
+        "onHeap"
+    ).in(LazySingleton.class);
+
+    PolyBind
+        .optionBinder(binder, Key.get(NamespaceExtractionCacheManager.class))
+        .addBinding("offHeap")
+        .to(OffHeapNamespaceExtractionCacheManager.class)
+        .in(LazySingleton.class);
+
+    getNamespaceFactoryMapBinder(binder)
+        .addBinding(JDBCExtractionNamespace.class)
+        .to(JDBCExtractionNamespaceCacheFactory.class)
+        .in(LazySingleton.class);
+    getNamespaceFactoryMapBinder(binder)
+        .addBinding(URIExtractionNamespace.class)
+        .to(URIExtractionNamespaceCacheFactory.class)
+        .in(LazySingleton.class);
+    getNamespaceFactoryMapBinder(binder)
+        .addBinding(StaticMapExtractionNamespace.class)
```

Comment:
This is defining a map at a specific level. It is not documented because it is not intended for use in anything other than workflow testing.